### ISSUE TITLE
Question about 3D spatial dimension order (D/H/W vs W/H/D) in PRIMUS

### ISSUE THREAD

**Issue by Yiming0131** [MIC-DKFZ/nnUNet#2962]
Hi! Thanks for releasing this great repo.

I have a question about the spatial dimension order used in Primus. In Pytorch, nn.Conv3d expects input tensors in (B, C, D, H, W), but i noticed some variable like: 

```
        FW, FH, FD = x.shape[2:]  # Full W , ...
        x = self.down_projection(x)
        # last output of the encoder is the input to EVA
        B, C, W, H, D = x.shape
```

This seems inconsitent with the Conv3d convention. Could you please clarify whether this different axis ordering is intentional? If so, could it affect downstream modules (e.g., positional/rotary embedding)?

Thanks a lot for your reply!

**Comment by TaWald** [comment]
Hey @Yiming0131, thanks for raising this.

Regarding the shapes in the primus repository, the naming convention is indeed inconsistent. 
However, as long as the `rearrange` operations are consistent with them it is not a functional issue. 

That being said, there used to be an issue in older versions of the `Primus` architecture, where the `rearrange` was messed up. (see this [commit](https://github.com/MIC-DKFZ/dynamic-network-architectures/commit/a2520df8b024b1fb720ba59bd8b359f8a2297657) in dynamic-network-architectures). 

But since this update in dynamic-network-architectures in June 2025 this issue should be fixed.

Long story short, this should not be a functional issue but just cosmetic.
If this bothers you too much, feel free to open a PR and I will ping Fabian to merge it. 

Best,
Tassilo 
  

P.S.: Soon, there should be a new SotA PrimusV3 version that will get released, so stay tuned in case you are interested.
